test/source/tests/unit-node.ts

```diff
@@ -844,15 +862,52 @@ jLwe8W9IMt765T5x5oux9MmPDXF05xHfm4qfH/BMO3a802x5u2gJjJjuknrFdgXY
       const plaintext = msgText
         .match(/Content\-Type: multipart\/mixed;\r?\n? boundary="\-\-\-\-sinikael\-\?=_2\-16054595384320\.6487848448108896".*\-\-\-\-\-\-sinikael\-\?=_2\-16054595384320\.6487848448108896\-\-\r?\n/s)![0]
         .replace(/\r?\n/g, '\r\n')!;
-      if ((await ContactStore.get(undefined, ['7FDE685548AEA788'])).length === 0) {
-        const contact = await ContactStore.obj({
-          email: 'flowcrypt.compatibility@gmail.com',
-          pubkey: testConstants.flowcryptcompatibilityPublicKey7FDE685548AEA788
+      const result = await MsgUtil.verifyDetached({
+        plaintext: Buf.fromUtfStr(plaintext),
+        sigText,
+        verificationPubs: [testConstants.flowcryptcompatibilityPublicKey7FDE685548AEA788]
+      });
+      expect(result.match).to.be.true;
+      t.pass();
+    });
+
+    ava.default(`[unit][MsgUtil.verifyDetached] returns non-fatal error when signature doesn't match`, async t => {
+      const sigText = Buf.fromUtfStr(`-----BEGIN PGP SIGNATURE-----
+
+wsD5BAABCAAjFiEE3CZFSvtx0Y6rutc9HH5tPFVjqUEFAl+QotYFAwAAAAAACgkQHH5tPFVjqUF4
+xgv+MrdQ07MfCVU93ptZg+S+OOkQ1AcZxGFdiivs10KkNGtLm9s+w/iEUAySSWbtKjbLV6O3AYvC
+QFKsFRFr17Ekz6mSPj99zifFMBvTOIAev/d08dmX0kGd6YlP+GyZL3Wqcgy1T1H3obgOmToDtk7R
+V52Ki1aTJYH/Z7v6PsQRWn8emfH/yGYplBhzZy2XjO6UIar9T8wtAJOd6+Ii2sfyGyEPjzGckLaR
+JZOxQ4jpJJUszz2WsvLNwtKoqwV15Eg3oxZzHWYE8P63xXoE4G762604SIqv/ggyQZTt/Es6Scun
+A1BJflFm+cHzQTW2yQfwCCvlzEZNiNwXfwGfV99K5iG1eW3lv7sMLJnitwTidNIlD5LTNdeUnTXj
+XJvkEQsyTUI4qbzzJbUNYz7lraizC2nPiwFzLv692mS0urtD3mUhOBA9hZwk3l/20GsGia0FeUIS
+E1d8Vh/Ey7IJ8TXbfFrdv5ZP3HqMK0089SooZwx/GN2QIaOYQXsS0u7IFNhU\n=q5Sf
+-----END PGP SIGNATURE-----`);
+      const data = await GoogleData.withInitializedData('flowcrypt.compatibility@gmail.com');
+      const msg = data.getMessage('1754cfc37886899e')!;
+      const msgText = Buf.fromBase64Str(msg!.raw!).toUtfStr();
+      {
+        const dhartleyPubkey = msgText
+          .match(/\-\-\-\-\-BEGIN PGP PUBLIC KEY BLOCK\-\-\-\-\-.*\-\-\-\-\-END PGP PUBLIC KEY BLOCK\-\-\-\-\-/s)![0]
+          .replace(/=\r\n/g, '').replace(/=3D/g, '=');
+        const resultRightKey = await MsgUtil.verifyDetached({
+          plaintext: Buf.fromUtfStr('some irrelevant text'),
+          sigText,
+          verificationPubs: [dhartleyPubkey]
         });
-        await ContactStore.save(undefined, contact);
+        expect(resultRightKey.match).to.be.null;
```

Comment:
Originally I imagine this would be a `false` ? Have the right pubkey, but didn't verify succcesfully.

Reply:
Fixed